packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -232,13 +233,15 @@ export function createComponent(_component, isAngleBracket, props, renderNode, e
     props._isAngleBracket = true;
   }
 
-  props.renderer = props.parentView ? props.parentView.renderer : env.container.lookup('renderer:-dom');
-  props._viewRegistry = props.parentView ? props.parentView._viewRegistry : env.container.lookup('-view-registry:main');
+  props.renderer = props.parentView ? props.parentView.renderer : env.owner.lookup('renderer:-dom');
+  props._viewRegistry = props.parentView ? props.parentView._viewRegistry : env.owner.lookup('-view-registry:main');
 
-  let component = _component.create(props);
+  const component = _component.create(props);
 
   // for the fallback case
-  component.container = component.container || env.container;
+  if (!getOwner(component)) {
```

Comment:
This is basically in few scenarios correct? We should be sure in nearly all cases owner is already set.